resim/metrics/default_report_metrics.py

```diff
@@ -0,0 +1,729 @@
+    )
+
+    fig = go.Figure(
+        data=[
+            go.Table(
+                header={"values": ["Metric Name", "Fail Count"]},
+                cells={
+                    "values": [
+                        [t[0] for t in job_metric_fail_counts_list],
+                        [t[1] for t in job_metric_fail_counts_list],
+                    ]
+                },
+            )
+        ]
+    )
+    fig.update_layout(
+        template="plotly_dark",
+        plot_bgcolor="rgba(0, 0, 0, 0)",
+        paper_bgcolor="rgba(0, 0, 0, 0)",
+    )
+
+    (
+        writer.add_plotly_metric("Failures per Job Metric")
+        .with_description("Job Metrics sorted by number of failures.")
+        .with_blocking(False)
+        .with_should_display(True)
+        .with_importance(mu.MetricImportance.HIGH_IMPORTANCE)
+        .with_status(mu.MetricStatus.PASSED_METRIC_STATUS)
+        .with_plotly_data(fig.to_json())
+    )
+
+    fig = go.Figure(
+        data=[
+            go.Table(
+                header={"values": ["Metric Name", "Fail Count"]},
+                cells={
+                    "values": [
+                        [t[0] for t in batch_metric_fail_counts_list],
+                        [t[1] for t in batch_metric_fail_counts_list],
+                    ]
+                },
+            )
+        ]
+    )
+    fig.update_layout(
+        template="plotly_dark",
+        plot_bgcolor="rgba(0, 0, 0, 0)",
+        paper_bgcolor="rgba(0, 0, 0, 0)",
+    )
+
+    (
+        writer.add_plotly_metric("Failures per Batch Metric")
+        .with_description("Batch Metrics sorted by number of failures.")
+        .with_blocking(False)
+        .with_should_display(True)
+        .with_importance(mu.MetricImportance.HIGH_IMPORTANCE)
+        .with_status(mu.MetricStatus.PASSED_METRIC_STATUS)
+        .with_plotly_data(fig.to_json())
+    )
+
+
+def batch_metrics_scalars_over_time_metric(
+    writer: ResimMetricsWriter,
+    *,
+    batches: list[Batch],
+    batch_to_jobs_map: dict[str, list[Job]],
+    job_to_metrics_map: dict[str, list[JobMetric]],
+    batch_to_metrics_map: dict[str, list[BatchMetric]],
+    scalar_batch_metrics_map: dict[str, dict[str, MetricProto]],
+) -> None:
+    # pylint: disable=unused-argument
+    """Creates metrics plotting batch metric scalars over time.
+
+    Creates a line plot showing the value over time for each scalar batch metric
+    defined by this report's test suite's metrics build.
+
+    Args:
+        writer: A metrics writer to write these metrics to.
+        batches: All batches for this report.
+        batch_to_jobs_map: All jobs for each batch.
+        job_to_metrics_map: All job metrics for each job.
+        batch_to_metrics_map: All batch metrics for each batch.
+        scalar_batch_metrics_map: All scalar batch metrics for each batch as protos.
+    """
+    values: dict[str, list] = defaultdict(list)
+    units: dict[str, str] = {}
+
+    for i, batch in enumerate(batches):
```

Comment:
foldall.map.map.filter :)

Reply:
I'm unfamiliar with this 😵‍💫 https://pypi.org/search/?q=foldall

Reply:
That was just a little functional programming joke :)